lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,528 @@
+            cr:save()
+            cr:set_source(color(bar_border_color))
+            cr:stroke()
+            cr:restore()
+        else
+            cr:stroke()
+        end
+    end
+
+    cr:translate(-x_offset, -y_offset)
+
+    -- Paint the handle
+    local handle_color = self._private.handle_color
+        or beautiful.slider_handle_color
+
+    -- It is ok if there is no color, it will be inherited
+    if handle_color then
+        cr:set_source(color(handle_color))
+    end
+
+    local handle_height, handle_width = height, self._private.handle_width
+        or beautiful.slider_handle_width
+        or height/2
+
+    local handle_shape = self._private.handle_shape
+        or beautiful.slider_handle_shape
+        or properties.handle_shape
+
+    -- Lets get the margins for the handle
+    margins = self._private.handle_margins
+        or beautiful.slider_handle_margins
+
+    x_offset, y_offset = 0, 0
+
+    if margins then
+        if type(margins) == "number" then
+            x_offset, y_offset = margins, margins
+            handle_width  = handle_width  - 2*margins
+            handle_height = handle_height - 2*margins
+        else
+            x_offset, y_offset = margins.left or 0, margins.top or 0
+            handle_width  = handle_width  -
+                (margins.left or 0) - (margins.right  or 0)
+            handle_height = handle_height -
+                (margins.top  or 0) - (margins.bottom or 0)
+        end
+    end
+
+    local value = self._private.value or self._private.min or 0
+
+    -- Get the widget size back to it's non-transfored value
+    --local matrix = gmatrix.from_cairo_matrix(cr:get_matrix())
+    local min, _, interval = get_extremums(self)
+    local rel_value = ((value-min)/interval) * (width-handle_width)
+
+    cr:translate(x_offset + rel_value, y_offset)
+
+    local handle_border_width = self._private.handle_border_width
+        or beautiful.slider_handle_border_width
+        or properties.handle_border_width or 0
+
+    handle_shape(cr, handle_width, handle_height)
+
+    if handle_border_width > 0 then
+        cr:fill_preserve()
+    else
+        cr:fill()
+    end
+
+    -- Draw the handle border
+    if handle_border_width > 0 then
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+local rect_order = {"x", "y", "width", "height"}
+
+-- Move the handle to the correct location
+local function move_handle(self, x, y, geo)
+    -- Fix the X position when used in a rotate or mirror widget
```

Comment:
Uhm, I'll take this as a bug report. x and y should already be in the widget's own space, but I see that the code in `drawable` for this is wrong (completely ignoring rotations). Do you want to fix this or should I do? Something like the following might work (untested):
```lua
diff --git a/lib/wibox/drawable.lua b/lib/wibox/drawable.lua
index 0212efa..6b450bb 100644
--- a/lib/wibox/drawable.lua
+++ b/lib/wibox/drawable.lua
@@ -376,8 +376,7 @@ function drawable.new(d, widget_context_skeleton, drawable_name)
             local widgets = ret:find_widgets(x, y)
             for _, v in pairs(widgets) do
                 -- Calculate x/y inside of the widget
-                local lx = x - v.x
-                local ly = y - v.y
+                local lx, ly = v.matrix_to_device:invert():transform_point(x, y)
                 v.widget:emit_signal(name, lx, ly, button, modifiers,v)
             end
         end)
```
(find_widgets should just be taught to also return the matrix_from_device so that this `invert()`-call isn't needed, but whatever...)